test/test_core.py

```diff
@@ -2257,6 +2257,10 @@ def test_em_asm_arguments_side_effects(self):
   def test_em_asm_direct(self):
     self.do_core_test('test_em_asm_direct.c')
 
+  def test_em_asm_side_module(self):
+    self.build(test_file('core/test_em_asm_side.c'), js_outfile=False, emcc_args=['-sSIDE_MODULE'], output_basename='side')
+    self.do_core_test('test_em_asm_main.c', emcc_args=['-sMAIN_MODULE=2', 'side.wasm'])
```

Comment:
is this now a core test because it's likely to be affected by optlevel? or just because we don't need to run it on multiple platforms?

Reply:
I wanted to make sure it passes at all opt levels (and in other configurations too).